# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsURIHashKey.h
# Commit: 86c478d114ec
# Full Hash: 86c478d114ec223ecf3698ddbb52977bf786f10c
# Author: Honza Bambas <honzab.moz@firemni.cz>
# Date: 2020-05-11 21:47:06
# Regressor Bug: 1618287
# File Overlap Count: 1
# Description:
#   Bug 1618287 - PreloaderBase class, r=smaug
#   
#   Depends on D71800
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D67479
# ==============================================================================

diff -r 5c64c1dfb318 -r 86c478d114ec netwerk/base/nsURIHashKey.h
--- a/netwerk/base/nsURIHashKey.h	Mon May 11 14:03:19 2020 +0000
+++ b/netwerk/base/nsURIHashKey.h	Mon May 11 14:20:39 2020 +0000
@@ -21,6 +21,7 @@
   typedef nsIURI* KeyType;
   typedef const nsIURI* KeyTypePointer;
 
+  nsURIHashKey() { MOZ_COUNT_CTOR(nsURIHashKey); }
   explicit nsURIHashKey(const nsIURI* aKey) : mKey(const_cast<nsIURI*>(aKey)) {
     MOZ_COUNT_CTOR(nsURIHashKey);
   }
@@ -30,6 +31,11 @@
   }
   MOZ_COUNTED_DTOR(nsURIHashKey)
 
+  nsURIHashKey& operator=(const nsURIHashKey& aOther) {
+    mKey = aOther.mKey;
+    return *this;
+  }
+
   nsIURI* GetKey() const { return mKey; }
 
   bool KeyEquals(const nsIURI* aKey) const {